src/Elcodi/Component/Page/Repository/PageRepository.php

```diff
@@ -29,30 +28,70 @@
  * @author Àlex Corretgé <alex@corretge.cat>
  * @author Berny Cantos <be@rny.cc>
  */
-class PageRepository extends EntityRepository implements PageRepositoryInterface
+class PageRepository extends EntityRepository
 {
     /**
-     * @param mixed $id
-     *
-     * @return PageInterface
-     */
-    public function findOneById($id)
-    {
-        return parent::find($id);
-    }
-
-    /**
      * Find one Page given its path
      *
      * @param string $path Page path
      *
-     * @return PageInterface
+     * @return PageInterface|null Page
      */
     public function findOneByPath($path)
     {
-        return parent::findOneBy(array(
+        return $this->findOneBy(array(
             'path'    => $path,
             'enabled' => true,
         ));
     }
+
+    /**
+     * Find pages paginated
+     *
+     * @param string  $type          Type
+     * @param integer $page          Page
+     * @param integer $numberPerPage Number per page
+     *
+     * @return array Pages
+     */
+    public function findPages($type, $page, $numberPerPage)
+    {
+        $offset = (($page - 1) * $numberPerPage) + 1;
```

Comment:
This is something the paginator should do, not the repository. Same for the next function… `getNumberOfEnabledPages`? Really?